### DISCUSSION TITLE
httpx[socks] + httpx[http2] ?

### DISCUSSION THREAD

**Discussion by seidnerj** [encode/httpx#2779] [Q&A] [answered]
How can I install httpx both with socks support as well as http2 support? are they mutually exclusive at the moment?

Thanks!

**Comment by karpetrosyan** [comment]
Hey!

You can install `httpx` like so

``` bash
$ pip install "httpx[http2, socks]"
```

> are they mutually exclusive at the moment?

Yes.

**Reply by seidnerj** [reply]
Thank you for your reply, I am a bit confused, you said they are mutually exclusive but also explained how to install both at the same time?

**Reply by karpetrosyan** [reply]
I believe I misunderstood you, but the answer is "yes" they can both be used.

**Reply by seidnerj** [reply]
Many thanks!

**Comment by seidnerj** [comment, marked as answer]
Just a side note to anyone else that is wondering, when installing this using pip from the command line either:
`pip install "httpx[http2, socks]"` or `pip install "httpx[http2,socks]"` are fine (the double quotes are mandatory though).

However, when including this dependency in a requirements.txt file, this should appear as `httpx[http2,socks]`, i.e. NO double quotes and NO spaces whatsoever. One can also include the version, as usual, like so: `httpx[http2,socks]==0.24.1`.
